widgetssdk/src/main/java/com/glia/widgets/core/secureconversations/SecureConversationsRepository.kt

```diff
@@ -69,11 +69,31 @@ internal class SecureConversationsRepository(private val core: GliaCore, private
 
     fun getUnreadMessagesCount(callback: RequestCallback<Int>) = secureConversations.getUnreadMessageCount(callback)
 
-    fun getHasPendingSecureConversations(): Single<Boolean> = Single.create { emitter ->
-        secureConversations.getPendingSecureConversationStatus { hasPendingSecureConversations, _ ->
-            if (!emitter.isDisposed) {
-                emitter.onSuccess(hasPendingSecureConversations ?: false)
+    fun subscribeToUnreadMessagesCount(callback: RequestCallback<Int>) = secureConversations.subscribeToUnreadMessageCount(callback)
+
+    fun unsubscribeFromUnreadMessagesCount(callback: RequestCallback<Int>) = secureConversations.unSubscribeFromUnreadMessageCount(callback)
+
+    fun unreadMessagesCountObservable(): Observable<Int> = Observable.create { emitter ->
+        val callback = RequestCallback<Int> { count, ex ->
+            when {
+                ex != null -> emitter.tryOnError(ex)
+                count != null -> emitter.onNext(count)
+                else -> emitter.onNext(NO_UNREAD_MESSAGES)
+            }
+        }
+        secureConversations.subscribeToUnreadMessageCount(callback)
+        emitter.setCancellable { secureConversations.unSubscribeFromUnreadMessageCount(callback) }
+    }
+
+    fun pendingSecureConversationsStatusObservable(): Observable<Boolean> = Observable.create { emitter ->
```

Comment:
NIT. I prefer a `getPendingSecureConversationsStatusObservable()` name.

Reply:
This kind of naming is more Java-specific, but I got a new idea regarding these functions.
In Kotlin, it is more idiomatic to use property access syntax for such functions. We can define a read-only property with a custom getter instead of using a function with the get prefix.

```kotlin
val pendingSecureConversationsStatusObservable: Observable<Boolean>
    get() = Observable.create { emitter ->
        val callback = RequestCallback<Boolean> { hasPendingSecureConversations, ex ->
            when {
                hasPendingSecureConversations != null -> emitter.onNext(hasPendingSecureConversations)
                ex != null -> emitter.tryOnError(ex)
                else -> emitter.onNext(false)
            }
        }
        secureConversations.subscribeToPendingSecureConversationStatus(callback)
        emitter.setCancellable { secureConversations.unSubscribeFromPendingSecureConversationStatus(callback) }
    }
``` 
And in Java, it will be automatically named with the `get` prefix.

Reply:
changed